sed/loader/fel/buffer.py

```diff
@@ -0,0 +1,243 @@
+
+                missing_in_parquet_str = (
+                    f"Missing in parquet: {missing_in_parquet}" if missing_in_parquet else ""
+                )
+                missing_in_config_str = (
+                    f"Missing in config: {missing_in_config}" if missing_in_config else ""
+                )
+
+                raise ValueError(
+                    "The available channels do not match the schema of file",
+                    f"{existing_parquet_filenames[i]}",
+                    f"{missing_in_parquet_str}",
+                    f"{missing_in_config_str}",
+                    "Please check the configuration file or set force_recreate to True.",
+                )
+
+    def get_files_to_read(
+        self,
+        h5_paths: list[Path],
+        folder: Path,
+        prefix: str,
+        suffix: str,
+        force_recreate: bool,
+    ) -> None:
+        """
+        Determines the list of files to read and the corresponding buffer files to create.
+
+        Args:
+            h5_paths (List[Path]): List of paths to H5 files.
+            folder (Path): Path to the folder for buffer files.
+            prefix (str): Prefix for buffer file names.
+            suffix (str): Suffix for buffer file names.
+            force_recreate (bool): Flag to force recreation of buffer files.
+        """
+        # Getting the paths of the buffer files, with subfolder as buffer and no extension
+        pq_handler = ParquetHandler(
+            [Path(h5_path).stem for h5_path in h5_paths],
+            folder,
+            "buffer",
+            prefix,
+            suffix,
+            extension="",
+        )
+        self.buffer_paths = pq_handler.parquet_paths
+        # read only the files that do not exist or if force_recreate is True
+        files_to_read = [
+            force_recreate or not parquet_path.exists() for parquet_path in self.buffer_paths
+        ]
+
+        # Get the list of H5 files to read and the corresponding buffer files to create
+        self.h5_to_create = list(compress(h5_paths, files_to_read))
+        self.buffer_to_create = list(compress(self.buffer_paths, files_to_read))
+
+        self.num_files = len(self.h5_to_create)
+
+        print(f"Reading files: {self.num_files} new files of {len(h5_paths)} total.")
+
+    def _create_buffer_file(self, h5_path: Path, parquet_path: Path) -> None:
+        """
+        Creates a single buffer file. Useful because h5py.File cannot be pickled if left open.
+
+        Args:
+            h5_path (Path): Path to the H5 file.
+            parquet_path (Path): Path to the buffer file.
+        """
+        # Open the h5 file in read mode
+        h5_file = h5py.File(h5_path, "r")
+
+        # Create a DataFrameCreator instance with the configuration and the h5 file
+        dfc = DataFrameCreator(self._config, h5_file)
+
+        # Get the DataFrame from the DataFrameCreator instance
+        df = dfc.df
```

Comment:
I tried that first but due to the parallel code, it fails to work. I forgot the error since it has been a long time.

Regarding the delayed, I'd like to try that too. I'll make an issue of it for after refactoring.

Reply:
I tested with dask.delayed but it seems since I open the h5 files, it doesnt really do that part lazily. So there's no performance gain. Perhaps I am using it in the wrong place.